### PR TITLE
add krocus

### PR DESCRIPTION
**Pull request by thanhleviet**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/krocus/krocus.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    <inputs>
+        <param name="fastq" type="data" format="fastqsanger,fastqsanger.gz" label="Select fastq dataset" help="Specify dataset with single read"/>
+        <param label="Select a database" name="krocus_mlst_databases" type="select">
+            <options from_data_table="krocus_mlst_databases">
+                <validator message="No database is available" type="no_options" />
+            </options>
+        </param>
+        <section name="opt_args" title="Optional Arguments">
+            <param argument="--filtered_reads_file" type="text" value="" optional="True" label="Filename to save matching reads to" help="Default: None" />
```
Comment (bgruening) [inline tools/krocus/krocus.xml]: The filename is not relevant for the user, this argument can be removed imho.

File: tools/krocus/krocus.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            <param argument="--max_gap" type="integer" value="4" label="Maximum gap for blocks to be contigous, measured in
```
Comment (bgruening) [inline tools/krocus/krocus.xml]: min="1"

File: tools/krocus/krocus.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                        multiples of the k-mer size" help="Default:4" />
+            <param argument="--margin" type="integer" value="10" label="Flanking region around a block to use for mapping" help="Default:10" />
```
Comment (bgruening) [inline tools/krocus/krocus.xml]: min="1"

File: tools/krocus/krocus.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            <param argument="--min_block_size" type="integer" value="150" label="Minimum block size in bases" help="Default:150" />
```
Comment (bgruening) [inline tools/krocus/krocus.xml]: min="1"

File: tools/krocus/krocus.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            <param argument="--min_fasta_hits" type="integer" value="10" label="Minimum No. of kmers matching a read" help="Default:10" />
```
Comment (bgruening) [inline tools/krocus/krocus.xml]: min="1"

File: tools/krocus/krocus.xml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            <param argument="--min_kmers_for_onex_pass" type="integer" value="5" label="Minimum No. of kmers matching a read in 1st pass" help="Default:5" />
+            <param argument="--max_kmers" type="integer" value="10" label="Dont count kmers occuring more than this many times" help="Default:10" />
+            <param argument="--print_interval" type="integer" value="10" label="Print ST every this number of reads" help="(Default:500)" />
+            <param argument="--kmer" type="integer" value="10" label="k-mer size" help="(Default:11)" />
+            <param argument="--divisible_by_3" type="boolean" truevalue="--divisible_by_3" falsevalue="" checked="false" label="Genes which are not divisible by 3 are excluded" help="(Default:No)" />
+            <param argument="--target_st" type="integer" value="" optional="True" label="For performance testing print time to find given ST" help="(Default:None)" />
```
Comment (bgruening) [inline tools/krocus/krocus.xml]: This parameter should also not be exposed to the user. A user does not test the performance and Galaxy is offering the runtime either way. So I think we can simplify the interface and remove this one :)

File: tools/krocus/krocus.xml (hunk 1 above)
Comment (thanhleviet) [reply tools/krocus/krocus.xml]: Thanks, removed

File: tools/krocus/krocus.xml (hunk 6 above)
Comment (thanhleviet) [reply tools/krocus/krocus.xml]: Removed.

**Comment by bgruening** [comment]
Thanks a lot @thanhleviet!

**Comment by bernt-matthias** [comment]
Dislike that the tool still does not use the default values... :(

**Comment by bernt-matthias** [comment]
Hi @thanhleviet . Due to my mistake my review comments were only displayed after @bgruening merged. We stopped the deployment of the tool to the TS via travis for the moment. 

If you have any other changes that you would like to implement you need to open a new PR against the updated master branch. If not I may also do this. 

Just tell me which way you prefer.

**Comment by thanhleviet** [comment]
> Hi @thanhleviet . Due to my mistake my review comments were only displayed after @bgruening merged. We stopped the deployment of the tool to the TS via travis for the moment.
> 
> If you have any other changes that you would like to implement you need to open a new PR against the updated master branch. If not I may also do this.
> 
> Just tell me which way you prefer.

Thanks @bernt-matthias , I can handle it myself.

**Comment by bernt-matthias** [comment]
Thanks a lot.

**Pull request merged**